app.go

```diff
@@ -496,49 +518,20 @@ func (app *app) runShell(s string, args []string, prefix string) cleanFunc {
 	var out io.Reader
 	var err error
 	switch prefix {
-	case "$|":
-		var stdin io.WriteCloser
-		stdin, err = cmd.StdinPipe()
-		if err != nil {
-			log.Printf("writing stdin: %s", err)
-		}
-		app.cmdIn = stdin
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
-
-		app.nav.previewChan <- ""
-		app.nav.dirPreviewChan <- nil
-
-		if err := app.ui.suspend(); err != nil {
-			log.Printf("suspend: %s", err)
-		}
-
-		err = cmd.Start()
 	case "$", "!":
 		cmd.Stdin = os.Stdin
 		cmd.Stdout = os.Stdout
 		cmd.Stderr = os.Stderr
 
-		app.nav.previewChan <- ""
-		app.nav.dirPreviewChan <- nil
-
-		if err := app.ui.suspend(); err != nil {
-			log.Printf("suspend: %s", err)
-		}
-		defer func() {
-			if err := app.ui.resume(); err != nil {
-				app.quit()
-				os.Exit(3)
-				return
-			}
-		}()
-		defer app.nav.renew()
+		app.runCmdSync(cmd, prefix == "!")
+		return
+	}
 
-		err = cmd.Run()
-	case "%":
-		shellSetPG(cmd)
+	// We are running the command asynchroniously
+	if prefix == "%" {
+		shellSetPG(cmd) // TODO: Delete or document the reason to do this twice for % mode
```

Comment:
I checked the `syscall` module documentation, this is used to give the child shell process its own process group (set the group process ID to its own process ID), so that you can kill the process group as a whole with `cmd-interrupt`.

See the code in https://github.com/gokcehan/lf/blob/88b3c999afb7cab8e8ca61da3c9063f0cc348c10/os.go#L149-L155

You shouldn't have to do this more than once, and I think this particular line can be deleted since `shellSetPG` is called later.

Reply:
Thanks for checking!